lib/foreman_rh_cloud.rb

```diff
@@ -14,4 +16,61 @@ def self.authentication_url
   def self.verify_ssl_method
     @verify_ssl_method ||= ENV['SATELLITE_RH_CLOUD_URL'] ? OpenSSL::SSL::VERIFY_NONE : OpenSSL::SSL::VERIFY_PEER
   end
+
+  def self.http_proxy_string
+    ForemanRhCloud.get_proxy
+  end
+
+  def self.transformed_http_proxy_string
+    ForemanRhCloud.transform_scheme(ForemanRhCloud.get_proxy)
+  end
+
+  def self.get_proxy
+    HttpProxy.default_global_content_proxy&.full_url ||
+    ForemanRhCloud.cdn_proxy ||
+    ForemanRhCloud.global_foreman_proxy
+  end
+
+  def self.cdn_proxy(is_proxy_scheme = true)
+    proxy_config = SETTINGS[:katello][:cdn_proxy]
+    return nil unless proxy_config
+
+    proxy_host = URI.parse(proxy_config[:host]).host
+    if proxy_host
+      proxy_port = proxy_config[:port]
+      uri = URI("#{scheme}://#{proxy_host}:#{proxy_port}")
+      if proxy_config[:user]
+        uri.user = CGI.escape(proxy_config[:user])
+        uri.password = CGI.escape(proxy_config[:password])
+      end
+
+      uri.to_s
+    end
+  rescue URI::Error => e
+    ForemanRhCloud.logger.warn("cdn_proxy parsing failed: #{e}")
+    nil
+  end
+
+  def self.global_foreman_proxy
+    Setting[:http_proxy]
+  end
+
+  # This method assumes uri_string contains uri-encoded username and p@$$word:
+  # http://user:p%40%24%24word@localhost:8888
+  def self.transform_scheme(uri_string)
+    transformed_uri = URI.parse(uri_string)
+
+    case transformed_uri.scheme
+    when "http"
+      transformed_uri.scheme = 'proxy'
+    when "https"
+      transformed_uri.scheme = 'proxys'
+    end
+
+    transformed_uri.to_s
+  end
+
+  def self.logger
+    Foreman::Logging.logger('background')
```

Comment:
You are in `lib` context, I don't think it's correct to use background logger specifically. I suppose it would be better to pass the logger into each method with a sane default:

``` ruby
  def self.cdn_proxy(is_proxy_scheme = true, logger: Foreman::Logging.logger('rh_cloud'))  
  #...
  rescue URI::Error => e
    logger.warn("cdn_proxy parsing failed: #{e}")
    nil
  end
```

Reply:
got the error: `RuntimeError: Trying to use logger rh_cloud which has not been configured.`

Reply:
Well then, we can use `app`. I think it looks for one of the options from [here](https://github.com/theforeman/foreman/blob/develop/config/settings.yaml.example#L78).